From ce8784be8f80af023f8eefd32b36390bc5d7bb50 Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Thu, 26 Feb 2015 11:42:28 +0100
Subject: [PATCH] DO NOT USE: test for BTRFS discard

This is just a small patch to see if BTRFS really discards everything
(I believe) it should. It won't kill your filesystem, it just overwrites
discarded blocks with zero instead of using a discard command the storage
might offer. But it might slow down your system and would lead to a maybe
shorter live of your SSD if you would use it all the time.
---
 fs/btrfs/extent-tree.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs/btrfs/extent-tree.c b/fs/btrfs/extent-tree.c
index a684086..f3ad7ba 100644
--- a/fs/btrfs/extent-tree.c
+++ b/fs/btrfs/extent-tree.c
@@ -1886,7 +1886,9 @@ static int remove_extent_backref(struct btrfs_trans_handle *trans,
 static int btrfs_issue_discard(struct block_device *bdev,
 				u64 start, u64 len)
 {
-	return blkdev_issue_discard(bdev, start >> 9, len >> 9, GFP_NOFS, 0);
+//	return blkdev_issue_discard(bdev, start >> 9, len >> 9, GFP_NOFS, 0);
+	pr_info("BTRFS: zeroout\n");
+	return blkdev_issue_zeroout(bdev, start >> 9, len >> 9, GFP_NOFS);
 }
 
 int btrfs_discard_extent(struct btrfs_root *root, u64 bytenr,
-- 
2.0.5

